modelexpress_client/python/modelexpress/gds_transfer.py

```
@@ -0,0 +1,245 @@
+        gpu_xfer = self._agent.get_xfer_descs(gpu_buffer, "VRAM")
+
+        # Execute transfer (READ = file → GPU)
+        handle = self._agent.initialize_xfer(
+            "READ", gpu_xfer, file_xfer, self._agent.name
+        )
+
+        state = self._agent.transfer(handle)
+        if state == "ERR":
+            self._agent.deregister_memory(file_descs)
+            self._agent.deregister_memory(gpu_descs)
+            raise RuntimeError(f"GDS transfer failed at offset {file_offset}")
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)